lib/syscall_shim/include/uk/syscall_entertab.h

```diff
@@ -0,0 +1,104 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#ifndef __UK_SYSENTERTAB_H__
+#define __UK_SYSENTERTAB_H__
+
+#include <uk/assert.h>
+
+struct uk_syscall_enter_ctx {
+	struct ukarch_execenv *execenv;
+};
+
+typedef void (*uk_syscall_entertab_func_t)(struct uk_syscall_enter_ctx *);
```

Comment:
Should we be returning a value in the entry handler in case we want to catch errors, similar to inittab?